tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java

```diff
@@ -79,4 +101,37 @@ private void assertDiagnostic(Diagnostic diagnostic, Object[] expected) {
         assertEquals(diagnostic.location().lineRange().startLine().line(), expected[1]);
         assertEquals(diagnostic.location().lineRange().startLine().offset(), expected[2]);
     }
+
+    private Object[][] getExpectedDiagnosticProperties() {
+        return new Object[][] {
+                {"[string]|record {| string arg2; |}", UNION},
+                {"()", NIL},
+        };
+    }
+
+    private void assertDiagnosticProperties(List<Diagnostic> diagnostics) {
+        int index = 0;
+
+        for (Diagnostic diagnostic : diagnostics) {
+
+            List<DiagnosticProperty<?>> diagnosticProperties = diagnostic.properties();
+
+            if (diagnosticProperties == null) {
+                continue;
+            }
+
+            for (DiagnosticProperty<?> property : diagnosticProperties) {
+
+                // The diagnostic property value is an instance of TypeSymbol only if the property kind is SYMBOLIC.
+                if (property.kind() == DiagnosticPropertyKind.SYMBOLIC) {
+                    TypeSymbol typeSymbol = (TypeSymbol) property.value();
+                    assertEquals(typeSymbol.signature(), getExpectedDiagnosticProperties()[index][0]);
+                    assertEquals(typeSymbol.typeKind(), getExpectedDiagnosticProperties()[index++][1]);
+                }
```

Comment:
Shall we remove these unnecessary new lines?

Reply:
Removed the superfluous newlines, except for the ones that are surrounding the if block.